lib/events/filesessions/fileasync.go

```diff
@@ -417,17 +417,27 @@ func (u *Uploader) startUpload(ctx context.Context, fileName string) error {
 		}
 	}
 
+	start := time.Now()
+	if err := u.takeSemaphore(ctx); err != nil {
+		return trace.Wrap(err)
+	}
+	if time.Since(start) > 500*time.Millisecond {
+		log.Debugf("Semaphore acquired in %v for upload %v.", time.Since(start), fileName)
+	}
+
 	// Apparently, exclusive lock can be obtained only in RDWR mode on NFS
 	sessionFile, err := os.OpenFile(sessionFilePath, os.O_RDWR, 0)
 	if err != nil {
+		_ = u.releaseSemaphore(ctx)
```

Comment:
Can we find a way to do this with defer? This seems like it will be way too easy to introduce a new branch in the future that doesn't release the semaphore.